docs/src/components/ButtonLink.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import styled, { css } from "styled-components";
+import { PolymorphicPropsWithoutRef } from "react-polymorphic-types";
+
+const DefaultComponent = "a";
+
+interface OwnProps {
+  type: "primary" | "secondary";
+  size?: "medium" | "large";
+  iconLeft?: React.ReactNode;
+  iconRight?: React.ReactNode;
+  children: React.ReactNode;
+}
+
+type Props<T extends React.ElementType = typeof DefaultComponent> = PolymorphicPropsWithoutRef<
+  OwnProps,
+  T
+>;
+
+const StyledComponent = styled(DefaultComponent)<Props>`
+  ${({ theme, type, size = "medium" }) => css`
+    display: inline-flex;
+    align-items: center;
+    ${size === "medium" &&
+    css`
+      padding: 0.875rem 1.5rem;
+    `}
+    ${size === "large" &&
+    css`
+      padding: 1.25rem 2rem;
+    `}
+    border-radius: 100px; /* value safely greater than button height to achieve pill effect */
+    font-weight: 500;
+
+    > * + * {
+      margin-left: 10px;
+    }
+
+    ${type === "primary" &&
```

Comment:
set `primary` as default and you can omit that condition, plus will not be needed to pass this prop, if type is not secondary.